From 79850e1025aabeccd3586c32ca259aa854607582 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Wed, 29 Sep 2021 17:38:32 +0000
Subject: [PATCH] Do not define tgmath.h fmaxmag, fminmag macros for C2X (bug
 28397)

C2X does not include fmaxmag and fminmag.  When I updated feature test
macro handling accordingly (commit
858045ad1c5ac1682288bbcb3676632b97a21ddf, "Update floating-point
feature test macro handling for C2X", included in 2.34), I missed
updating tgmath.h so it doesn't define the corresponding type-generic
macros unless __STDC_WANT_IEC_60559_BFP_EXT__ is defined; I've now
reported this as bug 28397.  Adjust the conditionals in tgmath.h
accordingly.

Tested for x86_64.
---
 math/tgmath.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/math/tgmath.h b/math/tgmath.h
index 9868353..3d7b099 100644
--- a/math/tgmath.h
+++ b/math/tgmath.h
@@ -921,7 +921,9 @@
 
 /* Like ilogb, but returning long int.  */
 # define llogb(Val) __TGMATH_UNARY_REAL_RET_ONLY (Val, llogb)
+#endif
 
+#if __GLIBC_USE (IEC_60559_BFP_EXT)
 /* Return value with maximum magnitude.  */
 # define fmaxmag(Val1, Val2) __TGMATH_BINARY_REAL_ONLY (Val1, Val2, fmaxmag)
 
-- 
1.8.3.1

